src/yea_wandb/mock_server.py

```diff
@@ -344,12 +349,101 @@ def to_dict(self):
+
+    _inject_count: int
+    _inject_time: float
+
+    def __init__(self):
+        self._inject_count = 0
+        self._inject_time = 0.0
+
+    def relay(self, func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+
+            # Normal mockserver mode, disable live relay and call next function
+            if not os.environ.get("MOCKSERVER_RELAY"):
+                return func(*args, **kwargs)
+
+            if request.method == "POST":
+                url_path = request.path
+                body = request.get_json()
+
+                url = f"https://api.wandb.ai{url_path}"
+                resp = requests.post(url, json = body)
+                data = resp.json()
+                run_obj = data.get("data", {}).get("upsertBucket", {}).get("bucket", {})
+                project_obj = run_obj.get("project", {})
+
+                run_id = run_obj.get("name")
+                project = project_obj.get("name")
+                entity = project_obj.get("entity", {}).get("name")
+
+                ctx = get_ctx()
+                if run_id:
+                    ctx["relay_run_info"].setdefault(run_id, {})
+                    ctx["relay_run_info"][run_id]["project"] = project
+                    ctx["relay_run_info"][run_id]["entity"] = entity
+
+                # TODO: This is a hardcoded for now, will add inject specification to the yea file
+                if run_id and run_id.startswith("inject"):
+                    time_now = time.time()
+                    if self._inject_count == 0:
+                        self._inject_time = time_now
+                    self._inject_count += 1
+                    if time_now < self._inject_time + 21:
+                        # print("INJECT", self._inject_count, time_now, self._inject_time)
+                        time.sleep(12)
+                        raise HttpException("some error", status_code=500)
+                return data
+            assert False
+
+            return func(*args, **kwargs)
+        return wrapper
+
+    def context_enrich(self, ctx):
+        for run_id, run_info in ctx["relay_run_info"].items():
+            run_num = len(ctx["runs"])
+            insert = run_id not in ctx["run_ids"]
+            if insert:
+                ctx["run_ids"].append(run_id)
+            run_ctx = ctx["runs"].setdefault(run_id, default_ctx())
+
+            # NOTE: not used, added for consistancy with non-relay mode
+            r = run_ctx.setdefault("run", {})
+            r.setdefault("display_name", f"relay_name-{run_num}")
+            r.setdefault("storage_id", "storageid{run_num}")
+            r.setdefault("project_name", "relay_proj")
+            r.setdefault("entity_name", "relay_entity")
+
+            # NOTE: We are using wandb but it isnt a strict depenedency
+            import wandb
+            api = wandb.Api()
+
+            # TODO: handle an error here
+            run = api.run(f"{run_info['entity']}/{run_info['project']}/{run_id}")
```

Comment:
Let's at least put it in a try/except clause and mark with exitcode=1 if it fails.

Reply:
add try/except.
im just skipping the run in the context for now.   exitcode=1 doesnt feel right, but need to think about this.